dojo/remote_user.py

```diff
@@ -0,0 +1,74 @@
+import logging
+from django.contrib.auth.middleware import RemoteUserMiddleware as OriginalRemoteUserMiddleware
+from django.contrib.auth.backends import RemoteUserBackend as OriginalRemoteUserBackend
+from netaddr import IPAddress
+from dojo.settings import settings
+from dojo.pipeline import assign_user_to_groups, cleanup_old_groups_for_user
+
+logger = logging.getLogger(__name__)
+
+
+class RemoteUserMiddleware(OriginalRemoteUserMiddleware):
+    def process_request(self, request):
+        # process only if request is comming from the trusted proxy node
+        if IPAddress(request.META['REMOTE_ADDR']) in settings.AUTH_REMOTEUSER_TRUSTED_PROXY:
+            self.header = settings.AUTH_REMOTEUSER_USERNAME_HEADER
+            return super().process_request(request)
+        else:
+            logger.debug('Requested came from untrusted proxy %s; This is list of trusted proxies: %s',
```

Comment:
I assume this should be error, not debug

Reply:
No. This is only for debugging purpose if somebody's request is not processed as expected.
If the request is coming from different IP, RemoteUser middleware is simply ignored and continues with another middleware.
The authentication can be performed by other configured middleware like SAML or standard Username+Password